sycl/source/event.cpp

```diff
@@ -117,7 +117,14 @@ event::get_profiling_info() const {
 
 #undef __SYCL_PARAM_TRAITS_SPEC
 
-backend event::get_backend() const noexcept { return getImplBackend(impl); }
+backend event::get_backend() const noexcept {
+  try {
+    return getImplBackend(impl);
+  } catch (std::exception &e) {
+    __SYCL_REPORT_EXCEPTION_TO_STREAM("exception in get_backend", e);
```

Comment:
This will be a user-visible message, so it should have more information

Reply:
The exception details comes from the origin (far down the call chain).
This function is to catch it and exit gracefully with whatever propagated from the exception origin.
There is no way at this outermost level can specify the details of the exception.